src/kernels/kernel.base.ts

```diff
@@ -138,7 +139,8 @@ export abstract class BaseKernel implements IKernel {
         public readonly controller: NotebookController,
         protected readonly configService: IConfigurationService,
         protected readonly workspaceService: IWorkspaceService,
-        readonly cellHashProviderFactory: CellHashProviderFactory
+        readonly cellHashProviderFactory: CellHashProviderFactory,
+        public readonly creator: KernelActionSource
```

Comment:
From this we know who owns a kernel (i.e. who created it, whether we or 3rd party extension).